thinknimble/tn-spa-bootstrapper
# Simpler Cypress Setup

## What this does

The current Cypress setup is too complicated. This change results in the same behavior with much less code and configuration.

## Checklist
- [x] Revert to the old simple way of running
- [x] Research ways to have Github actions pause until it detects a new deploy has occurred
- [x] Implement one of the approaches researched on another project
- [x] Confirm that Cypress isn't triggered on Build success, but on Release success
- [x] Test what happens when 2 commits are pushed back-to-back
- [ ] After merge, remove `HEROKU_API_KEY` and `GITHUB_TOKEN` from Github and Heroku respectively
- [ ] After merge, turn the Cypress merge requirement back on.

## Deploy Notes

n/a

## How to test

You will either need to bootstrap a new project or test this on an existing project (PR author has done the later)

Push a commit and observe that the required Chrome and Firefox steps are yellow.
After the release phase in Heroku, observe that Chrome and Firefox are triggered to run.
Push 2 more commits (spaced ~1min apart)
Observe that when the first release phase finishes, Chrome and Firefox are only triggered on the older of the two commits and that the PR is still blocked for merging.
Observe that when the 2nd release phase finishes, Cypress is finally triggered on the newest commit.

## Edge cases

Found only one minor edge case. If you push two commits back-to-back quickly, it's possible that when Cypress runs on the first commit that it will be testing the newer version of code. I think this is a very unlikely scenario to occur and, if it does occur, is unlikely to be noticed or matter to the dev team.
